finesse/gui/temp_control.py

```diff
@@ -41,138 +34,147 @@ def _create_controls(self) -> QGridLayout:
-        """
-        self._figure, self._ax1 = plt.subplots()
+        """Creates the matplotlib figure to be contained within the panel."""
+        self._figure, ax = plt.subplots()
+        self._ax = {"hot": ax}
         self._canvas = FigureCanvasQTAgg(self._figure)
 
-        t = [1040, 1050, 1060, 1070, 1080, 1090, 1100]
-        hot_bb_temp = [55, 57.5, 60, 62.5, 65, 67.5, 70]
-        cold_bb_temp = [1, 1.1, 1.15, 1.2, 1.3, 1.4, 1.5]
+        # Placeholder callback to test updating of figure
+        self._canvas.mpl_connect("button_press_event", self._update_figure)
+
+        self._figure_num_pts = 10
+        t = [None] * self._figure_num_pts
+        hot_bb_temp = [None] * self._figure_num_pts
+        cold_bb_temp = [None] * self._figure_num_pts
+
+        colours = plt.rcParams["axes.prop_cycle"]
+        hot_colour = colours.by_key()["color"][0]
+        cold_colour = colours.by_key()["color"][1]
 
-        self._bb_hot_line = self._ax1.plot(t, hot_bb_temp, color=[0, 1, 0])
-        self._ax1.set_xlabel("")
-        self._ax1.set_ylabel("HOT BB", color=[0, 1, 0])
-        self._ax1.set_xlim([1045, 1101.4])
-        self._ax1.set_ylim([20, 80])
+        self._ax["hot"].plot(t, hot_bb_temp, color=hot_colour)
+        self._ax["hot"].set_ylabel("HOT BB", color=hot_colour)
+        self._ax["hot"].set_xlim([1045, 1101.4])
+        self._ax["hot"].set_ylim([20, 80])
 
-        self._ax2 = self._ax1.twinx()
-        self._bb_cold_line = self._ax2.plot(t, cold_bb_temp, color=[1, 1, 0])
-        self._ax2.set_ylabel("COLD BB", color=[1, 1, 0])
-        self._ax2.set_ylim([0, 10])
+        self._ax["cold"] = self._ax["hot"].twinx()
+        self._ax["cold"].plot(t, cold_bb_temp, color=cold_colour)
+        self._ax["cold"].set_ylabel("COLD BB", color=cold_colour)
+        self._ax["cold"].set_ylim([0, 10])
 
         self._canvas.draw()
 
-    def _update_figure(self) -> None:  # , x, y) -> None:
-        """Updates the matplotlib figure to be contained within the panel.
+    def _btn_pressed(self, name) -> None:
+        """Shows or hides BB plots."""
+        state = self._ax[name].yaxis.get_visible()
+        self._btns[name].setFlat(state)
+        self._ax[name].yaxis.set_visible(not state)
+        self._ax[name].lines[0].set_visible(not state)
+        self._canvas.draw()
 
-        Args:
-            #x: time
-            #y: temperature
-            # probably won't be inputs since obtainable from elsewhere
-        Returns:
-            None
-        """
-        xdata = list(self._bb_hot_line[0].get_xdata())
-        y1data = list(self._bb_hot_line[0].get_ydata())
-        y2data = list(self._bb_cold_line[0].get_ydata())
+    def _update_figure(self, event) -> None:
+        """Updates the matplotlib figure to be contained within the panel."""
+        xdata = list(self._ax["hot"].lines[0].get_xdata())
+        y1data = list(self._ax["hot"].lines[0].get_ydata())
+        y2data = list(self._ax["cold"].lines[0].get_ydata())
```

Comment:
These variables could be given more descriptive names. How about `hot_data` and `cold_data`?